scalyr_agent/util.py

```diff
@@ -1269,3 +1272,87 @@ def close(self):
             """Closes the channel to the server.
             """
             pass
+
+
+def parse_array_of_strings(strlist):
+    """Convert comma-separated string list into an ArrayOfStrings
+
+    Accepts the following string representations.
+    ['a', 'b', 'c']
+    ["a", "b", "c"]
+    'a', 'b', 'c'
+    "a", "b", "c"
+    a, b, c
+
+    @param strlist: list to be converted
+    @return: None if strlist is empty, else return a JsonArray of strings
+    @raise TypeError if element_type is specified and conversion of any element fails
+    """
+    if not strlist:
+        return None
+    strlist = strlist.strip()
+
+    # Remove surrounding square brackets
+    if strlist[0] == '[' and strlist[-1] == ']':
+        strlist = strlist[1:-1]
+
+    # Extract elements, removing any surrounding quotes (Single-quotes are illegal JSON. Double quotes will be added).
+    elems = []
+    items = strlist.split(',')
+    for elem in items:
+        elem = elem.strip()
+        if len(elem) == 0:
+            continue
+        if elem[0] == r"'" or elem[0] == r'"' and elem[-1] == elem[0]:
+            elem = elem[1:-1]
+        if len(elem) == 0:
+            continue
+        elems.append(elem)
+
+    return ArrayOfStrings(*elems)
+
+
+def get_config_from_env(param_name, custom_env_name=None, convert_to=None):
+    """Returns the environment variable value for a config param
+
+    If a custom environment variable name is defined, use it instead of prepending 'SCALYR_'.
+
+    @param param_name: Config param name (may be global or module-level)
+    @custom_env_name: Custom environment variable name
+    @expected_type: if not None, will convert and validate to this type.  Otherwise, will leave as string.
+    @return: 3-tuple containing (param_name, env string val, converted value)
```

Comment:
Just personal preference.. returning a tuple like this is a little bit of a pain.  Do we return all of this just to support the error logging if there is a conflict in a config provided value?  If so, maybe we should move that logging here?  After all, we do have two instances of that logging.. it might be better to consolidate it.

Reply:
Consolidating logging sgtm. Will try.